internal/graph/extensions/live_query/json_patch.go

```diff
@@ -67,66 +69,41 @@ func (r *JSONPatchReporter) Report(rs cmp.Result) {
 	assert(len(r.path) > 0)
 	ps := r.path.Last()
 	vx, vy := ps.Values()
-	var op Operation
 	switch s := ps.(type) {
 	case cmp.MapIndex, cmp.TypeAssertion:
 		switch {
 		// value did not exist.
 		case !vx.IsValid():
-			op = Operation{
-				Op:    Add,
-				Path:  pathAsJSONPointer(r.path),
-				Value: vy.Interface(),
-			}
+			r.op(Add, r.toJSONPointer(r.path), vy.Interface(), "")
 		// value does not exist.
 		case !vy.IsValid():
-			op = Operation{
-				Op:   Remove,
-				Path: pathAsJSONPointer(r.path),
-			}
+			r.op(Remove, r.toJSONPointer(r.path), nil, "")
 		// value replaced.
 		default:
-			op = Operation{
-				Op:    Replace,
-				Path:  pathAsJSONPointer(r.path),
-				Value: vy.Interface(),
-			}
+			r.op(Replace, r.toJSONPointer(r.path), vy.Interface(), "")
 		}
 	case cmp.SliceIndex:
 		kx, ky := s.SplitKeys()
 		switch {
 		// value was updated.
 		case kx == ky:
-			op = Operation{
-				Op:    Replace,
-				Path:  pathAsJSONPointer(r.path),
-				Value: vy.Interface(),
-			}
+			r.op(Replace, r.toJSONPointer(r.path), vy.Interface(), "")
 		// value did not exist before.
 		case kx == -1:
-			op = Operation{
-				Op:    Add,
-				Path:  pathAsJSONPointer(r.path[:len(r.path)-1]) + "/" + strconv.Itoa(ky),
-				Value: vy.Interface(),
-			}
+			r.op(Add, r.toJSONPointer(r.path[:len(r.path)-1])+"/"+strconv.Itoa(ky), vy.Interface(), "")
 		// value was removed.
 		case ky == -1:
-			op = Operation{
-				Op:   Remove,
-				Path: pathAsJSONPointer(r.path[:len(r.path)-1]) + "/" + strconv.Itoa(kx),
-			}
+			r.op(Remove, r.toJSONPointer(r.path[:len(r.path)-1])+"/"+strconv.Itoa(kx), nil, "")
 		// value was moved.
 		default:
-			op = Operation{
-				Op:   Move,
-				Path: pathAsJSONPointer(r.path[:len(r.path)-1]) + "/" + strconv.Itoa(ky),
-				From: pathAsJSONPointer(r.path[:len(r.path)-1]) + "/" + strconv.Itoa(kx),
-			}
+			r.move(
+				r.toJSONPointer(r.path[:len(r.path)-1])+"/"+strconv.Itoa(kx),
+				r.toJSONPointer(r.path[:len(r.path)-1])+"/"+strconv.Itoa(kx),
+			)
```

Comment:
I'm probably misunderstanding something, but these paths are the same. Is one of them meant to reference `ky` rather than `kx`?

Reply:
yeah good catch @branden. I fixed it but since I no longer use go-cmp to generate patch, this isn't there anymore.